plugins/modules/idrac_diagnostics.py

```diff
@@ -0,0 +1,633 @@
+        export_status = self.__export_diagnostics(module, payload, export_diagnostics_url)
+        return export_status
+
+    def __get_export_diagnostics_url(self, module):
+        uri, error_msg = validate_and_get_first_resource_id_uri(
+            self.module, self.idrac, MANAGERS_URI)
+        if error_msg:
+            self.module.exit_json(msg=error_msg, failed=True)
+        resp = get_dynamic_uri(self.idrac, uri)
+        url = resp.get('Links', {}).get(OEM, {}).get(MANUFACTURER, {}).get(LC_SERVICE, {}).get('@odata.id', {})
+        action_resp = get_dynamic_uri(self.idrac, url)
+        export_url = action_resp.get(ACTIONS, {}).get(EXPORT, {}).get('target', {})
+        return export_url
+
+    def __export_diagnostics(self, module, payload, export_diagnostics_url):
+        diagnostics_file_name = module.params.get('share_parameters').get('file_name')
+        if not diagnostics_file_name:
+            now = datetime.now()
+            hostname = module.params.get('idrac_ip')
+            diagnostics_file_name = hostname + "_" + now.strftime("%Y%m%d_%H%M%S") + ".txt"
+        payload["FileName"] = diagnostics_file_name
+        diagnostics_status = self.idrac.invoke_request(export_diagnostics_url, "POST", data=payload)
+        return diagnostics_status
+
+    def __check_file_extension(self, module):
+        file_name = module.params.get('share_parameters').get('file_name')
+        if file_name:
+            file_extension = file_name.lower().endswith(".txt")
+            if not file_extension:
+                module.exit_json(msg=INVALID_FILE_MSG, failed=True)
+
+    def get_job_status(self, module, export_diagnostics_status):
+        res_uri = validate_and_get_first_resource_id_uri(self.module, self.idrac, MANAGERS_URI)
+        job_tracking_uri = export_diagnostics_status.headers.get("Location")
+        job_id = job_tracking_uri.split("/")[-1]
+        job_uri = IDRAC_JOB_URI.format(job_id=job_id, res_uri=res_uri[0])
+        job_failed, msg, job_dict, wait_time = idrac_redfish_job_tracking(self.idrac, job_uri)
+        job_dict = remove_key(job_dict, regex_pattern='(.*?)@odata')
+        if job_failed:
+            module.exit_json(msg=job_dict.get('Message'), failed=True, job_details=job_dict)
+        return job_dict
+
+    def get_share_details(self, module):
+        share_details = {}
+        share_details["IPAddress"] = module.params.get('share_parameters').get('ip_address')
+        share_details["ShareName"] = module.params.get('share_parameters').get('share_name')
+        share_details["UserName"] = module.params.get('share_parameters').get('username')
+        share_details["Password"] = module.params.get('share_parameters').get('password')
+        share_details["FileName"] = module.params.get('share_parameters').get('file_name')
+        return share_details
+
+    def get_proxy_details(self, module):
+        proxy_details = {}
+        if module.params.get('share_parameters').get('share_type') == "http":
```

Comment:
You can optimize the code as below and avoid ifelse
proxy_details["ShareType"] = self.module.params.get('share_parameters').get('share_type').upper()

Reply:
done